src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -271,6 +273,16 @@ public void setNamespace(String namespace) {
         this.namespace = Util.fixEmpty(namespace);
     }
 
+    public String getJnlpregistry() {
+        return jnlpregistry;
+    }
+
+    @DataBoundSetter
+    public void setJnlpregistry(String jnlpregistry) {
+        this.jnlpregistry = Util.fixEmpty(jnlpregistry);
+        PodTemplateBuilder.DEFAULT_JNLP_DOCKER_REGISTRY_PREFIX = jnlpregistry;
```

Comment:
This is clearly wrong, since the constant is static and would apply to every kubernetes cloud instance.

Reply:
The only thing to be added in the PR is a UI version of the configuration that can be done using `org.csanchez.jenkins.plugins.kubernetes.pipeline.PodTemplateStepExecution.dockerRegistryPrefix`. The system property would also be for every kubernetes cloud instance.

Did I understand the issue correctly, that we have the setting attached to a specific "cloud" in Jenkins instead of on a global level?

Reply:
Yes, if you add a UI setting, this applies to a single cloud instance, and need to be passed down from the cloud to the pod builder.
Likely in https://github.com/jenkinsci/kubernetes-plugin/blob/46150ead1f72265d847d85957966e903cd1bd916/src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java#L314-L318, you'd read the configuration from the cloud instance if available.

Reply:
Thanks @Vlatombe for this input. 👍 
I have made according changes in c7c590b and have verified that the JNLP image registry can be specified per cloud.